### DISCUSSION TITLE
[Run issue]: Larger history runs are now ending and writing scored optical photons or particles into files

### DISCUSSION THREAD

**Discussion by VictoriaJPR** [OpenTOPAS/OpenTOPAS#56] [Run-Problem]
### Contact Details

vjp12@duke.edu

### What was the issue?

Hey all! I am writing a program to score optical photons and particles (Energy Deposit, Optical Photons, Phase Space) in a water material with a specific molecule for a lab simulated experiment. I have a Workstation TS P8_PROM21_ES_TW_R, AMD Ryzen Threadripper PRO 7985WX {processor with 64GB DDR5 memory - so I don't think that is the problem. However, anything over 100 histories is not writing the files and finishing the simulation no matter how much time I give it to (I have let it go for 2 days). It can do the simulation itself with 100k histories in about 3 minutes, creates the ASCII, CSV files but never writes any data in it.  It seems to write the  ASCII files for optical photon scoring if I still kill the terminal after its done. Could someone help me out? Thank you!!

[main4.txt](https://github.com/user-attachments/files/16285647/main4.txt)

### TOPAS category

Scoring

### Version

3.7

### What Operative System (OS) are you running on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by icmeyer** [comment]
Thanks for posting and adding your input file! Not sure exactly what the issue is, but maybe something to do with the large number of threads or size of the resulting tallies leading to a memory issue. It looks like you have hundreds of phasespace tallies which is a lot! How big are your phasespace files after the 100 initial history run?

Some steps you could take to debug the issue (in order of what I think might work):
- Switch from ASCII to Binary output for the phasespace scorers
- Try manually setting the scorer buffer size with `i:Sc/MyScorer/OutputBufferSize = 1000`, could try something smaller
- Change `ShowHistoryCountAtInterval` to a larger number to reduce the printing to console

**Reply by icmeyer** [reply]
Not sure if this is relevant to your issue, but I just encountered a similar issue where the run completed but didn't finalize, and the offending bit of the input was a graphics setting. Maybe worth checking that you comment out all graphics parameters in all of your input files.

**Comment by JoseRamosMendez** [comment]
Hi there,
It's kind of hard to debug because the production of optical photons depends on the material's optical properties. And those are not included in your file. Have you tried the $TOPAS/examples/Optical/OpticalPhotonCount.txt? Please note that the optical photon parameter configuration changed for OpenTOPAS v4.0 see [here](https://opentopas.readthedocs.io/en/latest/parameters/physics/optical.html#optical-photons)
